library/CHART-TEMPLATE/templates/sidecar-agent-configmap.yaml

```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "common.naming.fullname" . }}-agent-config
+data:
+  newrelic-infra.yml: |
+    {{- $agentDefaults := fromYaml (include "common.agentConfig.defaults" . ) }}
+    {{- $agentSettings := .Values.agent | default dict }}
+    {{- mustMergeOverwrite $agentDefaults $agentSettings | toYaml | nindent 4 }}
```

Comment:
do we have a test covering the order of priority of each config?

Reply:
I completely forgot during this strange week. Sorry.
 
 I'm on it.

Reply:
There are two new commits: the first one with the tests and the second one fixing issues raised by the tests.

Everything should be solved with [#3bad27b](https://github.com/newrelic/helm-charts/pull/757/commits/3bad27b464c2a47b003ab924bc3b100c74ea30e3)